test_cases/test_cases.py

```diff
@@ -1,121 +1,31 @@
-         '      \'self\'\n      ;\n'
-         '    child-src\n'
-         '      \'self\'\n'
-         '      blob:\n      ;\n'
-         '    connect-src\n'
-         '      \'self\'\n      *\n      ;\n'
-         '    font-src\n'
-         '      \'self\'\n      ;\n'
-         '    frame-src\n'
-         '      \'self\'\n      ;\n'
-         '    img-src\n'
-         '      \'self\'\n'
-         '      data:\n      *\n      ;\n'
-         '    script-src\n'
-         '      \'self\'\n'
-         '      \'unsafe-eval\'\n'
-         '      \'unsafe-inline\'\n'
-         '      https://cdn.segment.com\n'
-         '      https://assets.customer.io\n'
-         '      https://secure.gaug.es\n'
-         '      https://www.google-analytics.com\n      ;\n'
-         '    style-src\n'
-         '      \'self\'\n'
-         '      \'unsafe-inline\'\n      ;\n  ">\n'
-         '  <script>\n'
-         '  if (/a\\.tiles\\.mapbox\\.com/.test(L.mapbox.config.HTTP_URL)) {\n'
-         '    Raven.config(\'https://c2d096c944dd4150ab7e44b0881b4a46@app.getsentry.com/11480\', {\n'
-         '      whitelistUrls: [/geojson\\.io/],\n'
-         '      ignoreErrors: [\n'
-         '          \'Uncaught Error: Error connecting to extension ckibcdccnfeookdmbahgiakhnjcddpki\',\n'
-         '          \'Uncaught Error: Error connecting to extension pioclpoplcdbaefihamjohnefbikjilc\'\n'
-         '      ]\n'
-         '    }).install();\n  }\n'
-         '  </script>\n'
-         '  <style>\n'
-         '  /*\n'
-         '   * http://seclab.stanford.edu/websec/framebusting/framebust.pdf\n'
-         '   */\n'
-         '  body { display: none; }\n'
-         '  </style>\n'
-         '</head>\n'
-         '<body id=\'geojsonio-body\'>\n\n'
-         '<div class=\'geojsonio\'></div>\n\n'
-         '<script>\n'
-         'if (self == top) {\n'
-         '  document.getElementsByTagName ("body")[0].style.display = \'block\';\n'
-         '} else {\n'
-         '  top.location = self.location;\n'
-         '}\n'
-         '</script>\n'
-         '<script src=\'dist/delegate.js\'></script>\n'
-         '<script src=\'dist/lib.js\'></script>\n'
-         '<script src=\'dist/site.js\'></script>\n'
-         '<script type=\'text/javascript\'>\n'
-         'if (/a\\.tiles\\.mapbox\\.com/.test(L.mapbox.config.HTTP_URL)) {\n'
-         '  var _gauges = _gauges || [];\n'
-         '  (function(i,s,o,g,r,a,m){i[\'GoogleAnalyticsObject\']=r;i[r]=i[r]||function(){\n'
-         '  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),\n'
-         '  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)\n'
-         '  })(window,document,\'script\',\'https://www.google-analytics.com/analytics.js\',\'ga\');\n\n'
-         '  ga(\'create\', \'UA-12158002-18\', \'auto\');\n'
-         '  ga(\'send\', \'pageview\');\n}\n'
-         '</script>\n'
-         '</body>\n'
-         '</html>\n'
+    xsd=open('../test_cases/valid_case.xsd', 'r')
```

Comment:
Could you change xsd as file to xsd_path and hide in case object open(...)?

Reply:
Something like that?

`class Case(object):
    def __init__(
            self, url_params, status_code, headers, xsd
    ):
        self.url_parametes = url_params
        self.exp_status_code = status_code
        self.exp_headers = headers
        self.xsd = open(xsd_path, 'r')
........
    xsd_path='../test_cases/valid_case.xsd'`

Reply:
Yes. Also you could read about with statement. https://docs.python.org/3/reference/compound_stmts.html#the-with-statement https://docs.python.org/2/tutorial/inputoutput.html

> It is good practice to use the with keyword when dealing with file objects. This has the advantage that the file is properly closed after its suite finishes, even if an exception is raised on the way. It is also much shorter than writing equivalent try-finally blocks: